flintrock/templates/spark/conf/spark-defaults.conf

```diff
@@ -0,0 +1 @@
+spark.jars.packages    org.apache.hadoop:hadoop-aws:{hadoop_version}
```

Comment:
You're right, `--packages` loads the transitive dependencies automatically. Thanks for pointing it ! According to [Spark documentation](https://spark.apache.org/docs/latest/submitting-applications.html#advanced-dependency-management) : 
```
Users may also include any other dependencies by supplying a comma-delimited list of maven coordinates with --packages. All transitive dependencies will be handled when using this command.
```

I tried the following with Spark 2.1, and it works : 
```
spark-shell --packages "org.apache.hadoop:hadoop-aws:2.7.3"

scala> spark.read.csv("s3a://bucket/pathTo/file.csv")

scala> res0.show
```